core/src/entry_verification_service/shared_wrapper.rs

```diff
@@ -0,0 +1,73 @@
+//! A simple wrapper for state shared between threads, that can also be waited on.
```

Comment:
This feels like it could be its own standalone crate. I don't see anything in here very specific to the entry verification service

Reply:
True.  But I didn't want to generalize too early.
So far, I'm writing what is needed for this service.
I didn't see this pattern in the code that I've seen so far, so I wrote it.
If another service will needed it, then it can be adjusted and moved into a shared location.
My personal experience is that it makes sense to write 3 specific versions, and only then to generalize them.  As, often, a generalization that comes from a single case is far from ideal.
But if you think I should put it into a shared location, just let me know.